nomad/csi_endpoint.go

```diff
@@ -251,18 +254,14 @@ func (v *CSIVolume) Deregister(args *structs.CSIVolumeDeregisterRequest, reply *
 		return structs.ErrPermissionDenied
 	}
 
-	state := v.srv.State()
-	index, err := state.LatestIndex()
+	_, index, err := v.srv.raftApply(structs.CSIVolumeDeregisterRequestType, args)
 	if err != nil {
+		v.logger.Error("csi raft apply failed", "error", err, "method", "deregister")
 		return err
 	}
 
-	err = state.CSIVolumeDeregister(index, args.VolumeIDs)
-	if err != nil {
-		return err
-	}
-
-	return v.srv.replySetIndex(csiVolumeTable, &reply.QueryMeta)
+	reply.Index = index
```

Comment:
We still need to call `srv.setQueryMeta(reply)` at some point here just as `replySetIndex` did, as we set a few other values there. (Plus we do that for all other RPC and if some day it has bigger side-effects we'd miss them here.)